code/datums/components/greentext.dm

```diff
@@ -0,0 +1,50 @@
+/datum/component/greentext
+	var/mob/living/last_holder = null
+	var/mob/living/current_holder
+	var/list/color_altered_mobs = list()
```

Comment:
This needs to be a list of weakrefs.

Also, make a `/Destroy()` that will set all 3 of these variables to null.